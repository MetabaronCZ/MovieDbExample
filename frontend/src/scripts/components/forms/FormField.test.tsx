import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { FormField } from './FormField';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/forms/FormField', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <FormField id="ID" label="LABEL">
        CONTENT
      </FormField>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    const label = getByText(container, 'LABEL');
    expect(label).toBeInTheDocument();
    expect(label).toHaveAttribute('for', 'ID');
  });

  it('should render vertical variant', async () => {
    const { container } = await waitForComponent(
      <FormField label="LABEL" orientation="vertical">
        CONTENT
      </FormField>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();
  });
});
