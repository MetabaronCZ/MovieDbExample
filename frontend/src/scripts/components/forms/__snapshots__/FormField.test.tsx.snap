// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/FormField should render properly 1`] = `
.c2 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

.c0 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  align-items: flex-start;
  justify-content: start;
}

<div>
  <fieldset
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="ID"
      >
        LABEL
      </label>
      CONTENT
    </div>
  </fieldset>
</div>
`;

exports[`components/forms/FormField should render vertical variant 1`] = `
.c2 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

.c0 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: none;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  align-items: flex-start;
  justify-content: start;
}

<div>
  <fieldset
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
      >
        LABEL
      </label>
      CONTENT
    </div>
  </fieldset>
</div>
`;
